rcv/frame_handlers/frc_2019.py

```diff
@@ -0,0 +1,240 @@
+            # and larger than 2950
+            if not (340 <= contour_area <= 2950):
+                continue
+            # get a parameter for how close the approximated should be
+            peri = cv2.arcLength(c, True)
+            # get the approximated contour shape for our contour
+            # each point of the approximated shape is closer then 2nd argument
+            # 0.02 is there because it works better on our videos
+            approx = cv2.approxPolyDP(c, 0.02 * peri, True)
+            # if the contour has more then 6 vertexes or less than 4 skip it
+            if not(4 <= len(approx) <= 6):
+                continue
+            # get the circumscribing rectangle
+            rect = cv2.minAreaRect(c)
+            box = cv2.boxPoints(rect)
+            box = np.int0(box)
+            # get the area of circumscribing rectangle area
+            box_area = cv2.contourArea(box)
+            # get the moments which will help us find the center of mass
+            # of the contour
+            M = cv2.moments(approx)
+            # calculate x,y coordinate of center
+            if M["m00"] != 0:
+                cY = int(M["m01"] / M["m00"])
+                cX = int(M["m10"] / M["m00"])
+            else:
+                cY = 0
+                cX = 0
+            half_x = self._frame.shape[1] / 2
+            # if the y of the center of mass of the new contour is bigger
+            # it means the contour is closer to us and is probably what we
+            # are looking for
+            if cY > fitting_mass_center[1]:
+                # if the current contour is closer in y and x to the bottom center
+                # it is out best fitting contour
+                if abs(half_x - cX) < abs(half_x - fitting_mass_center[0]):
+                    cont = box
+                    fitting_mass_center = (cX, cY)
+                # if the current contour has bigger y then the last one
+                # by an amount that matters it is probably what we are
+                # looking for
+                elif cY - fitting_mass_center[1] > 20:
+                    cont = box
+                    fitting_mass_center = (cX, cY)
+            # if the contour is closer in x to the center but doesn't have bigger Y
+            elif abs(half_x - cX) < abs(half_x - fitting_mass_center[0]):
+                # then we check if the difference in Y matters and if it does
+                # it is probably what are looking for
+                if cY - fitting_mass_center[1] > -20:
+                    cont = box
+                    fitting_mass_center = (cX, cY)
+        return cont
+
+    def clean_frame(self):
+        # get a gray copy of frame
+        gray = cv2.cvtColor(self._frame, cv2.COLOR_BGR2GRAY)
+        focused_image = cv2.bitwise_and(gray, self._focus_image_gray)
+        # use threshold on frame
+        _, thresh = cv2.threshold(focused_image, 230, 255, cv2.THRESH_BINARY)
```

Comment:
I think maybe the threshold should be given in the configuration to make it easier to use

Reply:
o.k.